content/raw/chat/username_avatar.md

```diff
@@ -0,0 +1,132 @@
+
+## Definitions
+
+- 3 word usernames - A 3 word username is a deterministic sequence of words that uniquely identifies a profile. see [spec](https://specs.status.im/spec/2#3-word-pseudonym--whisperwaku-key-fingerprint)
+- display name - A username choosen by the user to be displayed in the chat.
+- local username - An alias for a profile choosen by the User. Only visible to the user.
+- verified identity - A profile that went through the Identity Verification process. See Identity Verification Spec.
+- profile emblem - an icon next to a profile that is used to qualify that profile.
+- emoji hash -  A deterministic sequence of emojis that uniquely identifies a profile.
+- identicon ring - A deterministic sequence of colors around a profile avar that is used to differentiate it from other profiles.
+
+## Designs
+
+- [designs](https://www.figma.com/file/IPpvkpDWabBKJTeo6bFop0/Kuba%E2%8E%9CDesktop?node-id=1837%3A335126)
+- [identicon ring](https://www.figma.com/file/IPpvkpDWabBKJTeo6bFop0/Kuba%E2%8E%9CDesktop?node-id=1837%3A335130)
+
+## Use Cases
+
+### Setting up a profile
+
+prequesites: User is on onboarding process or has already set up a profile and has choosen to rename their username.
+
+1. User fills their intended display name
+2. User presses Save or Next
+3. System saves the new display name
+3.1 alternative: System warns user that the display name is already taken in one of the communities the user is part of
+3.2 alternative: System warns user that the display name is invalid
+
+### Joining a community
+
+1. User selects a community to join
+2. System detects that another user with the same display name has already joined that community
+3. System warns user that the display name is already taken in that community
+4. System displays conflicting profiles display name, avatar, identicon ring, emoji hash and chat key
+5. User selects "Change Your display name"
+6. User fills in a new display name
+7. Systems adds user to the community
+7.1 alternative: System warns user that the display name is already taken in one of the communities the user is part of
+7.2 alternative: System warns user that the display name is invalid
+
+## Functional Requirements
+
+**Profile Emblems:**
+
+- If a profile is a mutual contact then it MUST display a mutual contact emblem (e.g a profile icon)
+- If a profile is a verified identity then it MUST display a verified emblem (e.g a tick icon)
+- If a profile is marked as untrusthworthy then it MUST display a untrustworthy emblem (e.g a exclamation mark icon)
+- If a profile is a mutual contact then it MAY display another emblem
+- If a profile is not a mutual contact then it MUST NOT display other emblems
+
+**Profile usernames**
+
+- Users MUST be able to set a display name for their own profile
+- Display names MUST be at least 5 characters long
+- Display names MUST be composed of alphanumeric characters, underscores, and/or hyphens
```

Comment:
According to the designs, the display names must allow spaces. Perhaps this could be changed to 
```suggestion
- Display names MUST be composed of alphanumeric characters, underscores, spaces and/or hyphens, but they cannot contain spaces at the beggining or end 
```